File: javas/Pajak.java
import java.util.Scanner;

public class Pajak {
    public static void main(String[] args) {
        System.out.println("Masukkan penghasilan: ");
        try (Scanner sc = new Scanner(System.in)) {
            long p = sc.nextLong(), s = p;
            double l1, l2 = 0, l3 = 0, l4 = 0;

            if (s > 500_000_000) { l4 = (s - 500_000_000) * 0.30; s = 500_000_000; }
            if (s > 250_000_000) { l3 = (s - 250_000_000) * 0.25; s = 250_000_000; }
            if (s > 50_000_000)  { l2 = (s - 50_000_000)  * 0.15; s = 50_000_000; }

            l1 = s * 0.05;
            double total = l1 + l2 + l3 + l4;

            System.out.println("Lapisan 1: " + l1);
            System.out.println("Lapisan 2: " + l2);
            System.out.println("Lapisan 3: " + l3);
            System.out.println("Lapisan 4: " + l4);
            System.out.println("Total Pajak: " + total);
        }
    }
}

File: pythons/pajak.py
penghasilan = int(input("Masukkan penghasilan: ")) 
# saya jadikan  variabel tersendiri agar lebih mudah untuk memahami kode secara keseluruhan
# pun ketika akan dikembangkan lagi, data inputan masih tersimpan, sehingga tidak perlu mengubah kode yang lain, cukup mengubah variabel penghasilan saja

sisa = penghasilan
total = 0

if sisa > 500_000_000:
    lap4 = (sisa - 500_000_000) * 0.30
    sisa = 500_000_000
else:
    lap4 = 0

if sisa > 250_000_000:
    lap3 = (sisa - 250_000_000) * 0.25
    sisa = 250_000_000
else:
    lap3 = 0

if sisa > 50_000_000:
    lap2 = (sisa - 50_000_000) * 0.15
    sisa = 50_000_000
else:
    lap2 = 0

lap1 = sisa * 0.05

total = lap1 + lap2 + lap3 + lap4

print("Lapisan 1:", lap1)
print("Lapisan 2:", lap2)
print("Lapisan 3:", lap3)
print("Lapisan 4:", lap4)
print("Total Pajak:", total)